### ISSUE TITLE
works - but no playout on apple tablets and smartphones

### ISSUE THREAD

**Issue by marcuslauer** [ndunand/moodle-filter_html5avtomp4#7]
Dear Nicolas,
thank you for your work. We support more than 900 schools at homeschooling at the moment - nonprofit. Our teachers use Atto editor and WebRTC for creating feedback for the pupils (short video or voice messages). Problem is that the iPads don't like the WebM file sources. 

All iPad/iPhone-Browser are showing an error at the video-frame: "X no compatible source".

We found your plugin and our Moodle Admin installed it. And it works. After cronjob-run we can find two files. The ogg and the new m4a (Audio) or the WebM and new mp4 (Video).

So I took my iPad for a new test. Audio - perfect now it works :-)
But the Video-Files don't work. But now - and I think this is one step further - we get an videoframe and a "play-button" in it. But when I click on the  play-button the video doesn't start.

What tip can i give our experts? Does ffmpeg have to convert to another (Apple compatible) output format? Do you know how to adjust that?

**Comment by marcuslauer** [comment]
An addition:
As I understand it, the second source is missing in the html code. When I open ATTO editor and klick at the "files" button I can see that the plugin created a second file, the mp4 version. So every thing is fine.

When I switch from view mode to html mode I see the following:

```
<video controls="">
  <source src="https://lms.bildung-rp.de/admintest2/draftfile.php/74/user/draft/945774334/704426678301831-video.webm">
  https://lms.bildung-rp.de/admintest2/draftfile.php/74/user/draft/945774334/704426678301831-video.webm
</video>
<br><p><br></p>
```

So I have no source to the new file, correct?

**Comment by ndunand** [comment]
Hello @marcuslauer ,

First thanks for your feedback and giving some context. It helps to show the code we put in the public domain is useful to others.

First, one think is that the audio files are converted immediately (because the conversion is usually really quick), whereas the video files are processed by a scheduled task( cron job). This should be why you have a difference in behavious between audio files (work on iPad) and video files (don't work on iPad) : the scheduled task may have not run properly.

As per your second comment, it is correct that you won't see a second mp4 source in the atto editor in source mode (because the source HTML is left intact). You have however seen that in the Atto "area files" the mp4 files is present – in this the case for both the video and the audio cases?

**Comment by marcuslauer** [comment]
yes, i can see both in the area files: for video: webm and mp4 version and audio: ogg and m4a version.

And yes, I see no difference at the code in Atto.

Should the plugin also add a second source to the html code (automatically) when the cronjob has finished?

**Comment by marcuslauer** [comment]
I only talk about the video stuff. Audio works perfect - this is solved for me :-)

**Comment by ndunand** [comment]
No difference should be displayed in the html source code in atto at all.

However, when you display the page, the filter should modify the HTML on the fly, and if you check the source code in your browser you should see two sources in both cases (in both the audio case and the video case). Is that the case?

**Comment by johnnylupino** [comment]
Hi guys,

Thank you for you hard work.

I can confirm that I am also experiencing the same issue with Moodle 3.8.8 and 3.8.5 versions (talking about the video conversion/recording)

The video conversion takes place and both files can be seen under Manage Files in Atto editor, however the mp4 is not used.
I have tested it with Safari and Chrome on iOS devices (iPhone and iPad), same result.


Archie

**Comment by ndunand** [comment]
Hi Archie, would it be possible to make a publicly available page where we can experience the issue?

**Comment by johnnylupino** [comment]
Hi Nicolas,

Unfortunately, I may not be able to do that. However, please check the error message that I see when trying to play it on iPhone. The other screenshot is from the file picker on Desktop PC, which I believe shows that the original webm file gets a new converted version.

Archie

![mp4-webm](https://user-images.githubusercontent.com/7383318/113742916-63cce300-9703-11eb-83a6-ac846b728886.png)
![html5tomp4](https://user-images.githubusercontent.com/7383318/113743176-aa224200-9703-11eb-9a03-f2eee386db23.png)

**Comment by ndunand** [comment]
@johnnylupino ,

The file picker showing the converted mp4 file is good. To understand why the iPhone does not "see" it I would need the raw rendered HTML from that page. See for instance https://www.browserstack.com/guide/how-to-debug-on-iphone on how to do this.
